### ISSUE TITLE
Unable to get Unifi Camera G4 Pro to record to HomeKit and streaming to HomeKit times out after 30 seconds

### ISSUE THREAD

**Issue by tazmdevil** [koush/scrypted#228]
At the moment, I have a single camera and UNVR implementation. The camera is a Unifi Camera G4 Pro and try as I might, it will not record to HomeKit. It will stream, but the stream times out after 30 seconds with the following error in the log:

`[HomeKit]: HomeKit Streaming RTCP timed out. Terminating Streaming.`

I have tried it both with the Unifi Protect Plugin and setting the camera up as a RSTP camera with the RTSP Plugin and the same behavior happens with both setups.

Once added to HomeKit, I set it to record when any motion is detected, and to notify when any motion is detected. I'll get the motion notification along with a screen capture in the notification, but no recording. When motion is detected, the following appears in the camera's console:

```
event {
  type: 'motion',
  start: 1650594135043,
  score: 0,
  smartDetectTypes: [],
  smartDetectEvents: [],
  metadata: {},
  camera: '624fd0060304df03e700049f',
  partition: null,
  user: null,
  id: '6262115a00867903e7006dcc',
  modelKey: 'event'
}
update packet {
  lastMotion: 1650594135043,
  isMotionDetected: true,
  eventStats: {
    motion: {
      today: 171,
      average: 109,
      lastDays: [Array],
      recentHours: [Array]
    },
    smart: { today: 0, average: 0, lastDays: [Array] }
  },
  recordingSchedules: [],
  isRecording: true
}
[HomeKit]: snapshot requested for reason: 1
update packet {
  lastMotion: 1650594146376,
  isMotionDetected: false,
  recordingSchedules: [],
  isRecording: true
}
update packet { lastMotion: 1650594145709, recordingSchedules: [] }
```

I have tried all of the recommended troubleshooting steps in the HomeKit plugin's README. I even set up a second iCloud account to make very sure it wasn't a problem with my original iCloud account and repeated the troubleshooting steps.

The UNVR is running UniFi OS Version 2.3.16 and Protect version 1.21.5.
The G4 Pro's firmware is 4.49.9.
Scypted 0.0.171 is running in the official Docker container with host networking.
The plugin versions are

| Name | Version |
| --- | --- |
| HomeKit | 0.0.243 |
| RTSP Camera Plugin | 0.0.51 |
| Rebroadcast Plugin | 0.1.224 |
| Snapshot Plugin | 0.0.27 |
| Unifi Protect Plugin | 0.0.109 |

**Comment by koush** [comment]
The timeout indicates some sort of network issue, because HomeKit is not sending RTCP packets back. That is probably why recordings aren't working either (though G4 Pro should use the medium stream for recording). Join Discord and provide a full console log of a streaming attempt.

**Comment by tazmdevil** [comment]
Yeah, you're right. I was having the problem with scrypted running on a RHEL host along with homebridge and a number of other docker apps. I had firewalld set up on the host. I got to looking and it appears rtp/rtcp uses random udp ports. I checked active ports with and without an active streaming session, and the session opens ~6 random udp ports on the host. So instead of poking a bunch of holes in my host's firewall, I moved scrypted to a raspberry pi, configured it just like it was on my RHEL host, and it works now. I don't know that a pi is my long term solution, but it's working for now.

Thanks for pointing me in the right direction.
